tests/test_kwargs_and_defaults.py

```diff
@@ -175,13 +175,26 @@ def test_positional_only_args(msg):
         m.pos_kw_only_mix(1, 2, 3)
     assert "incompatible function arguments" in str(excinfo.value)
 
+    with pytest.raises(TypeError) as excinfo:
+        m.pos_only_def_mix()
+    assert "incompatible function arguments" in str(excinfo.value)
+
+    assert m.pos_only_def_mix(1) == (1, 2, 3)
+    assert m.pos_only_def_mix(1, 4) == (1, 4, 3)
+    assert m.pos_only_def_mix(1, 4, 7) == (1, 4, 7)
```

Comment:
How does this line work? If `k` is keyword-only, shouldn't this one throw?

Reply:
It's not:
```cpp
py::arg("i"), py::arg("j") = 2, py::pos_only(), py::arg("k") = 3);
```
First is pos only, second is pos only with default, and third is either pos or keyword, with default.